privacyidea/lib/eventhandler/base.py

```diff
@@ -356,10 +398,19 @@ def check_condition(self, options):
                 if not ','.join([tok.get_serial() for tok in token_objects]):
                     return False
 
-        if "tokenrealm" in conditions and tokenrealms:
+        if CONDITION.TOKENREALM in conditions and tokenrealms:
             res = False
             for trealm in tokenrealms:
-                if trealm in conditions.get("tokenrealm").split(","):
+                if trealm in conditions.get(CONDITION.TOKENREALM).split(","):
```

Comment:
An idea: We could get rid of ``res`` and change this to 
```python
for trealm in tokenrealms:
    if trealm in conditions.get(CONDITION.TOKENREALM).split(","):
        break
else:
    return False
```
because Python executes the ``else`` clause only if the loop exited normally, i.e. it is not executed if a ``break`` was encountered.
But the ``for...else`` syntax is a bit obscure, so I'm okay with keeping this like it is. What do you think?

Reply:
Personally i find this a bit difficult to read/understand. I just don't expect an `else:` here and for me it implies, that the loop did **not** finish correctly.

Reply:
...so when will the ``else`` branch be called?
I look at this and I do not unterstand it ;-)

Reply:
The ``else`` branch is executed only if we did *not* break out of the loop -- if you're used to it, it's quite useful, but I agree that it's not very readable. So let's keep ``res`` :-)